### PR TITLE
`children.replace` function

### PR DESCRIPTION
**Pull request by adamjanicki2**
Addresses #36, and is required to implement #33

### CODE REVIEW

File: src/children.js [hunk 1]
```diff
@@ -34,6 +36,28 @@ export function paths (node) {
...
 
+/**
+ * Replaces a child node with a new node, and updates the parent node and parent pointers
+ * @param {object} child 
+ * @param {object} newChild 
+ */
+export function replace (child, newChild) {
+	const parentPath = parents.path(child);
+	if (parentPath) {
+		const {property, index, node: parent} = parentPath;
```
Comment (LeaVerou) [inline src/children.js:47]: This will error if `parentPath` is `null` or `undefined`

Comment (adamjanicki2) [reply src/children.js:47]: How will it error? If it is null or undefined then the node is a root or doesn't have parents set, meaning replacement is not possible, what were you thinking should happen in this case?

File: src/children.js [hunk 2]
```diff
@@ -34,6 +36,36 @@ export function paths (node) {
...
+ */
+export function replace (child, newChild) {
+	const parentPath = parents.path(child);
+	if (parentPath === undefined) {
+		throw new Error("The child node does not have a parent node set");
+	}
+
+	// A root node was passed in
+	if (parentPath === null) {
+		return;
```
Comment (LeaVerou) [inline src/children.js:53]:
```suggestion
		// TODO warn("Replacing a root node does change anything");
		return;
```

File: src/children.js [hunk 3]
```diff
@@ -34,6 +36,36 @@ export function paths (node) {
...
+
+	if (index !== undefined) {
+		parent[property][index] = newChild;
+	}
+	else {
+		parent[property] = newChild;
+	}
+
+	parents.clear(child);
+	parents.set(newChild, parentPath, {force: true});
```
Comment (LeaVerou) [inline src/children.js:66]:
```suggestion
	parents.set(newChild, parentPath, {force: true});
	return newChild;
```

File: src/children.js [hunk 4]
```diff
@@ -34,6 +36,36 @@ export function paths (node) {
...
+/**
+ * Replaces a child node with a new node, and updates the parent node and parent pointers
+ * @param {object} child
+ * @param {object} newChild
+ * @throws {Error} If the child node does not have a parent node set
+ */
+export function replace (child, newChild) {
+	const parentPath = parents.path(child);
+	if (parentPath === undefined) {
+		throw new Error("The child node does not have a parent node set");
```
Comment (LeaVerou) [inline src/children.js:48]:
```suggestion
		throw new Error("Cannot replace a child node with no parent pointer. Call parents.set() on the node or parents.setAll() on an ancestor to add parent pointers to this node");
```

**Review by LeaVerou — APPROVED** [review]

**Pull request merged**